libraries/botbuilder-dialogs-adaptive-testing/src/setTestOptionsMiddleware.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @module botbuilder-dialogs-adaptive-testing
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Middleware, TurnContext, ActivityTypes } from 'botbuilder-core';
+
+const CONVERSATION_STATE = 'ConversationState';
+
+export class SetTestOptionsMiddleware implements Middleware {
+    /**
+     * Processes an incoming event activity.
+     * @param context The context object for this turn.
+     * @param next The delegate to call to continue the bot middleware pipeline
+     */
+    public async onTurn(context: TurnContext, next: () => Promise<void>): Promise<void> {
+        if (context.activity.type === ActivityTypes.Event) {
+            if (context.activity.name === 'SetTestOptions') {
+                const conversationState = context.turnState.get(CONVERSATION_STATE);
+                const property = conversationState.createProperty('TestOptions');
+                await property.set(context, context.activity.value);
+            }
+        }
+        if (next) {
```

Comment:
Is `next` allowed to be falsy? If so, make sure to change the method signature to account for this.